### PR TITLE
Update docs and chart metadata

### PR DESCRIPTION
**Pull request by mcavoyk**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Other notes for your reviewer**:

**Release note**:

```release-note
Update helm chart to install CRD's by default
```

### CODE REVIEW

File: deploy/charts/secret-manager/Chart.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: secret-manager
 description: External secret management for Kubernetes
 type: application
 version: 0.1.2
-appVersion: 0.0.0
+appVersion: "%%APP_VERSION%%"
```
Comment (DirtyCajunRice) [inline deploy/charts/secret-manager/Chart.yaml:6]: The chart will not lint like this. It needs to be semver compatible which is why it was 0.0.0

File: deploy/charts/secret-manager/Chart.yaml [hunk 2]
```diff
@@ -37,3 +36,12 @@ annotations:
       email: nick@cajun.pro
     - name: Kellin McAvoy
       email: kellinmcavoy@gmail.com
+  artifacthub.io/operator: "true"
+  artifacthub.io/operatorCapabilities: "Basic Install"
+  artifacthub.io/images: |
+    - name: Dockerhub Image
```
Comment (DirtyCajunRice) [inline deploy/charts/secret-manager/Chart.yaml:42]: Adding Image to the image name is redundant

File: .github/workflows/helm.yaml [hunk 3]
```diff
@@ -25,7 +25,7 @@ jobs:
       - name: Fix appversion for tests
         run: |
           LASTTAG=$(git describe --abbrev=0 --tags)
-          sed -i "s/0.0.0/${LASTTAG#v}/" deploy/charts/secret-manager/Chart.yaml
+          sed -i "s/%%APP_VERSION%%/${LASTTAG#v}/" deploy/charts/secret-manager/Chart.yaml
```
Comment (DirtyCajunRice) [inline .github/workflows/helm.yaml:28]: see comment about semver

File: .github/workflows/helm.yaml [hunk 4]
```diff
@@ -78,7 +78,7 @@ jobs:
           make crds-to-chart
           LASTTAG=$(git describe --abbrev=0 --tags)
           echo "Using ${LASTTAG#v} for appVersion"
-          sed -i "s/0.0.0/${LASTTAG#v}/" deploy/charts/secret-manager/Chart.yaml
+          sed -i "s/%%APP_VERSION%%/${LASTTAG#v}/" deploy/charts/secret-manager/Chart.yaml
```
Comment (DirtyCajunRice) [inline .github/workflows/helm.yaml:81]: same with semver here

**Review by DirtyCajunRice — CHANGES_REQUESTED** [review]

File: deploy/charts/secret-manager/Chart.yaml (hunk 1 above)
Comment (mcavoyk) [reply deploy/charts/secret-manager/Chart.yaml:6]: The workflow fixes this before lint and test, which is why it passed on this branch. Also locally, `helm lint` doesn't seem to mind this, only `version` is checked against semver, `appVersion` is just checked if it is a string ([source](https://github.com/helm/helm/blob/e1d4c674375e724d6cf72c91219986cc453f4c76/pkg/lint/rules/chartfile.go#L59))

**Review by DirtyCajunRice — APPROVED** [review]

**Comment by sbaier1** [comment]
just a heads-up: this PR removed the only documentation/examples on the `template` field in the CRD. It should be reworked/moved into the docs/ directory IMHO.

**Pull request merged**